dev/breeze/tests/test_docker_command_utils.py

```diff
@@ -0,0 +1,173 @@
+    mock_check_docker_permission.return_value = False
+    mock_run_command.return_value.returncode = 0
+    mock_run_command.return_value.stdout = "20.10.0"
+    check_docker_version(verbose=True)
+    mock_check_docker_permission.assert_called_with(True)
+    mock_run_command.assert_called_with(
+        ['docker', 'version', '--format', '{{.Client.Version}}'],
+        verbose=True,
+        suppress_console_print=True,
+        capture_output=True,
+        text=True,
+    )
+    mock_console.print.assert_called_with("Good version of Docker: 20.10.0.")
+
+
+@mock.patch('airflow_breeze.utils.docker_command_utils.check_docker_permission')
+@mock.patch('airflow_breeze.utils.docker_command_utils.run_command')
+@mock.patch('airflow_breeze.utils.docker_command_utils.console')
+def test_check_docker_version_higher(mock_console, mock_run_command, mock_check_docker_permission):
+    mock_check_docker_permission.return_value = False
+    mock_run_command.return_value.returncode = 0
+    mock_run_command.return_value.stdout = "21.10.0"
+    check_docker_version(verbose=True)
+    mock_check_docker_permission.assert_called_with(True)
+    mock_run_command.assert_called_with(
+        ['docker', 'version', '--format', '{{.Client.Version}}'],
+        verbose=True,
+        suppress_console_print=True,
+        capture_output=True,
+        text=True,
+    )
+    mock_console.print.assert_called_with("Good version of Docker: 21.10.0.")
+
+
+@mock.patch('airflow_breeze.utils.docker_command_utils.run_command')
+@mock.patch('airflow_breeze.utils.docker_command_utils.console')
+def test_check_docker_compose_version_unknown(mock_console, mock_run_command):
+    check_docker_compose_version(verbose=True)
+    expected_run_command_calls = [
+        call(
+            ["docker-compose", "--version"],
+            verbose=True,
+            suppress_console_print=True,
+            capture_output=True,
+            text=True,
+        ),
+    ]
+    mock_run_command.assert_has_calls(expected_run_command_calls)
+    mock_console.print.assert_called_with(
+        'Unknown docker-compose version. At least 1.29 is needed! \
+        If Breeze fails upgrade to latest available docker-compose version'
+    )
+
+
+@mock.patch('airflow_breeze.utils.docker_command_utils.run_command')
+@mock.patch('airflow_breeze.utils.docker_command_utils.console')
+def test_check_docker_compose_version_low(mock_console, mock_run_command):
+    mock_run_command.return_value.returncode = 0
+    mock_run_command.return_value.stdout = "1.28.5"
+    check_docker_compose_version(verbose=True)
+    mock_run_command.assert_called_with(
+        ["docker-compose", "--version"],
+        verbose=True,
+        suppress_console_print=True,
+        capture_output=True,
+        text=True,
+    )
+    
+    # mock_console.print.assert_called_with('You have too old version of docker-compose: 1.28.5!                 At least 1.29 is needed! Please upgrade!')
```

Comment:
@potiuk I have one more doubt in this mock part. we have two print statement if the docker compose version is low. But in the mock i could verify only one. Do you have any idea how i can add both prints in the mock?

Reply:
You can use `assert_has_calls` - you can find similar examples in airflow code :)